src/main/java/io/aiven/kafka/connect/opensearch/DocumentIDStrategy.java

```diff
@@ -0,0 +1,153 @@
+         record -> null,
+         (request, record) -> Objects.isNull(record.value()) ? null : request
+    ),
+    
+    RECORD_KEY("record.key", "Generated from the record's key",
+        record -> RecordConverter.convertKey(record.keySchema(), record.key()),
+        (request, record) ->  {
+            request.versionType(VersionType.EXTERNAL);
+            request.version(record.kafkaOffset());
+            return request;
+        }
+    ),
+
+    TOPIC_PARTITION_OFFSET("topic.partition.offset", "Generated as record's ``topic+partition+offset``",
+        record ->  record.topic() + "+" + record.kafkaPartition() + "+" + record.kafkaOffset(),
+        (request, record) -> request
+    );
+
+            
+    private final String name;
+
+    private final String description;
+
+    private final Function<SinkRecord, String> docIdGenerator;
+
+    private final BiFunction<DocWriteRequest<?>, SinkRecord, DocWriteRequest<?>> updateRequest;
+
+    private DocumentIDStrategy(final String name, final String description, 
+                               final Function<SinkRecord, String> docIdGenerator,
+                               final BiFunction<DocWriteRequest<?>, SinkRecord, DocWriteRequest<?>> updateRequest) {
+        this.name = name.toLowerCase(Locale.ROOT);
+        this.description = description;
+        this.docIdGenerator = docIdGenerator;
+        this.updateRequest = updateRequest;
+    }
+
+    @Override
+    public String toString() {
+        return name;
+    }
+
+    public static DocumentIDStrategy fromString(final String name) {
+        for (final DocumentIDStrategy strategy : DocumentIDStrategy.values()) {
+            if (strategy.nameEquals(name)) {
+                return strategy;
+            }
+        }
+        throw new IllegalArgumentException("Unknown Document ID Strategy " + name);
+    }
+
+    public static String[] names() {
+        return Arrays.stream(values()).map(v -> v.toString()).toArray(String[]::new);
+    }
+
+    public static String describe() {
+        return Arrays.stream(values()).map(v -> v.toString() + " : " + v.description)
+                .collect(Collectors.joining(", ", "{", "}"));
+    }
+
+    public Boolean nameEquals(final String name) {
+        return name.equalsIgnoreCase(this.name);
+    }
+
+    /**
+     * Return the DOC ID for a record using this Strategy
+     * @param record The record for which to generate the DOC ID
+     * @return Doc ID
+     */
+    public String docId(final SinkRecord record) {
+        return docIdGenerator.apply(record);
+    }
+
+    /**
+     * Update an IndexRequest for a record to include its DOC ID and External
+     * Version if applicable based on this Strategy.
+     * @param request The IndexRequest to be updated
+     * @param record The record for which to update the IndexRequest
+     * @return Updated IndexRequest
+     */
+    public DocWriteRequest<?> updateIndexRequest(final IndexRequest request,
```

Comment:
Please just `updateRequest`, overloaded methods for  `IndexRequest` and `DeleteRequest`